src/Components/ElementsPanel/Internal/VisualRuleElementsList.tsx

```diff
@@ -0,0 +1,274 @@
+        const elementItemWithColorings: ICardboardGroupedListItem<ITwinToObjectMapping> = {
+            id: rowId,
+            ariaLabel: element.displayName,
+            buttonProps: {
+                customStyles: buttonStyles.elementButton,
+                ...(onItemHover && {
+                    onMouseEnter: () => onItemHover(element, panelItem),
+                    onFocus: () => onItemHover(element, panelItem)
+                }),
+                ...(onItemBlur && {
+                    onMouseLeave: () => onItemBlur(element, panelItem),
+                    onBlur: () => onItemBlur(element, panelItem)
+                })
+            },
+            iconStart: () => (
+                <ElementColoring
+                    rowId={rowId}
+                    colorings={elementColorings}
+                    isModal={isModal}
+                />
+            ),
+            item: element,
+            itemType: 'header',
+            onClick: () => onItemClick(element, panelItem),
+            textPrimary: element.displayName
+        };
+        listItems.push(elementItemWithColorings);
+
+        badges.map((badge) => {
+            const badgeStyles = getElementsPanelAlertStyles(badge.visual.color);
+            const onEnter =
+                onItemHover && (() => onItemHover(element, panelItem));
+            const onLeave =
+                onItemBlur && (() => onItemBlur(element, panelItem));
+            const alertItem: ICardboardGroupedListItem<IExpressionRangeVisual> = {
+                ariaLabel: badge.visualRuleDisplayTitle,
+                buttonProps: {
+                    customStyles: buttonStyles.alertButton,
+                    onMouseEnter: onEnter,
+                    onFocus: onEnter,
+                    onMouseLeave: onLeave,
+                    onBlur: onLeave
+                },
+                iconStart: () => (
+                    <span className={badgeStyles.alertCircle}>
+                        <Icon iconName={badge.visual.iconName} />
+                    </span>
+                ),
+                item: badge.visualRule,
+                itemType: 'item',
+                onClick: () =>
+                    onItemClick(badge.visualRule, panelItem, badge.behavior),
+                textPrimary: badge.visualRuleDisplayTitle
+            };
+            listItems.push(alertItem);
+        });
+    });
+    return listItems;
+}
+
+interface IElementColoringProps {
+    colorings: VisualColorings[];
+    isModal: boolean;
+    rowId: string;
+}
+
+const ElementColoring: React.FC<IElementColoringProps> = (props) => {
+    const tooltipId = useId('cb-element-coloring-header-tooltip');
+    const theme = useExtendedTheme();
+
+    if (!props.isModal) {
+        const tooltipContent = (colorings: VisualColorings[]) => {
+            return <ColorPillsTooltip visualColorings={colorings} />;
+        };
+
+        return (
+            <TooltipHost
```

Comment:
don't you have a ColorPillTooltip component? Why do we need this copied here as well as somewhere else earlier in the PR? They seem like nearly copies.

Reply:
ColorPillTooltip is the content of the tooltip, I can add Content to the name if that makes it clearer.

Reply:
I'll also remove this Tooltip host in favor of the callout + onHover of Cardboardlist item